File: IntersectionOf2Arrays2.py
class Solution:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        
        a = {}
        
        b = {}
        
        result = []
        
        
        for i in nums1:
            
            if i in a:
                
                a[i] = a[i] + 1
            
            else:
                
                a[i] = 1
        
        
        for j in nums2:
            
            if j in b:
                
                b[j] = b[j] + 1
                
            else:
                
                b[j] = 1
        
        
        
        for key, value in a.items():
            
            if key in b:
                
                for i in range(0, min(value, b[key])):
                    
                    result.append(key)
        
        
        return result
        
        

File: IntersectionOf2Arrays2.cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) 
    {
        int size = nums1.size();
        
        unordered_map<int, int> map;
        
        for(int i = 0; i < size; i++)
        {
            map[nums1[i]] = map[nums1[i]] + 1;
        }
        
        
        size = nums2.size();
        
        vector<int> result;
        
        for(int i = 0; i < size; i++)
        {
            if(map.find(nums2[i]) != map.end())
            {
                if(map[nums2[i]] == 1)
                {
                    result.push_back(nums2[i]);
                    map.erase(nums2[i]);
                }
                
                if(map[nums2[i]] > 1)
                {
                    result.push_back(nums2[i]);
                    map[nums2[i]] = map[nums2[i]] - 1;
                }
            }
        }
        
        return result;
    }
};